ruby-challenges.rb

```diff
@@ -2,29 +2,150 @@
+#     Whenever you call the method new on a class, as in method.new, the class will create a new instance of itself. It will then, internally, call the method initialize on the new object. Doing so it will simply pass all the arguments that you passed to new on to the method initialize.   
+#     So we can now create a new Bike instance by calling …
+#     Bike.new("Trek")
+#     … and the string "Trek" will be passed on to our initialize method, and end up being assigned to the local variable .
+class Bike
+    attr_accessor :model, :wheels, :current_speed
+   def initialize(model,wheels,current_speed)
+    @model = model
+    @wheels = wheels
+    @current_speed = current_speed
+   end
+   def bike_info
+    puts "The #{@model} has #{@wheels}  and is going #{@current_speed}"
+end
+end
+
+
+# my_bike = Bike.new('Trek','2 wheels','0 mph')
+
+print my_bike.bike_info
+# output: The Trek has 2 wheels  and is going 0 mph
+
+
+# (attempt 1)class Bike
+#     def initialize (model,wheels,current_speed)
+#         @model = model
+#         @wheels = wheels
+#         @current_speed = current_speed
+#     end
+# end
+
+
+# # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
+
 
 
-# --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
+# -------------------3b) Add the ability to pedal faster and brake. The pedal_faster method should increase the speed by a given amount. The brake method should decrease the speed by a given amount. The bike cannot go negative speeds. Pseduocode: What do I want to do ? add the ability to pedal faster and brake to my Bike Class. How will I do this ? I can create two methods to add to the Bike class, the pedal_faster method will increase the speed by a given amount. the brake method will decrease the speed by an given amount. The methods will probably affect the current_speed instance variable?
 
-# Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
+# Ok I'm offically stumped about how to go about solving this challenge, I believe the numbers are causing me the majority of my grienvences. I don't really know how to manipulate the actual syntax to create the pedal_faster and brake method.  Defintely need a refactor, I'm unable to keep the numbers from going into the negatives when calling upon the brake method.
+class Bike
+    attr_accessor :model, :wheels, :current_speed,
+   def initialize(model,wheels,current_speed)
+    @model = model
+    @wheels = wheels
+    @current_speed = current_speed
+   end
+ def pedal_faster(new_speed)
+    @new_speed = new_speed
```

Comment:
You are really close here with the pedal_faster and brake methods! You want to make changes to your current speed so I would take the new_speed param and add or subtract from the current speed accordingly.